src/server/qgswmsserver.cpp

```diff
@@ -2079,6 +2079,15 @@ int QgsWMSServer::configureMapRender( const QPaintDevice* paintDevice ) const
     mMapRenderer->setOutputUnits( QgsMapRenderer::Pixels ); //SLD units are in pixels normally
   }
 
+  //Clear expression context and set project variables
+  QgsRenderContext* ctx = mMapRenderer->rendererContext();
+  if ( ctx )
+  {
+    QgsExpressionContext newContext;
+    newContext.appendScope( QgsExpressionContextUtils::projectScope() );
```

Comment:
@mhugent I'd make sense to first add the global scope here, then things like the platform name variable would be available. That could be useful for people who want to tweak the appearance of layers differently between desktop and server